docs/source/advanced/multi-task-learning.rst

```diff
@@ -0,0 +1,266 @@
+
+.. code-blocks:: yaml
+
+    ---
+    version: 1.0
+    gsf:
+        basic:
+            ...
+        ...
+        multi_task_learning:
+            - node_classification:
+                target_ntype: "movie"
+                label_field: "label"
+                mask_fields:
+                    - "train_mask_field_nc"
+                    - "val_mask_field_nc"
+                    - "test_mask_field_nc"
+                task_weight: 1.0
+            - edge_classification:
+                target_etype:
+                    - "user,rating,movie"
+                label_field: "rate"
+                mask_fields:
+                    - "train_mask_field_ec"
+                    - "val_mask_field_ec"
+                    - "test_mask_field_ec"
+                task_weight: 0.5 # weight of the task
+
+Task specific hyperparameters in multi-task learning are same as those for single task learning as
+detailed in :ref:`Training and Inference<configurations-run>`, except that two new configs are required,
+i.e., `mask_fields` and `task_weight`. The `mask_fields` provides the specific training, validation and
+test masks for a task and the `task_weight` gives its loss weight.
+
+In multi-task learning, GraphStorm provides a new unsupervised training signal, i.e., node feature
+reconstruction (`BUILTIN_TASK_RECONSTRUCT_NODE_FEAT = "reconstruct_node_feat"`). You can define a
+node feature reconstruction task as following:
+
+.. code-blocks:: yaml
+
+    ---
+    version: 1.0
+    gsf:
+        basic:
+            ...
+        ...
+        multi_task_learning:
+            - node_classification:
+                ...
+            - reconstruct_node_feat:
+                reconstruct_nfeat_name: "title"
+                target_ntype: "movie"
+                batch_size: 128
+                mask_fields:
+                    - "train_mask_c0" # node classification mask 0
+                    - "val_mask_c0"
+                    - "test_mask_c0"
+                task_weight: 1.0
+                eval_metric:
+                    - "mse"
+
+In the configuration, `target_ntype`` defines the target node type, the reconstruct node feature
+learning will be applied. `reconstruct_nfeat_name`` defines the name of the feature to be
+re-construct. The other configs are same as node regression tasks.
+
+
+Run Model Training
+~~~~~~~~~~~~~~~~~~~
+GraphStorm introduces a new command line `graphstorm.run.gs_multi_task_learning` to run multi-task learning tasks. You can use the following command to start a multi-task training
+job:
+
+.. code-block:: bash
+
+    python -m graphstorm.run.gs_multi_task_learning \
+              --workspace <PATH_TO_WORKSPACE> \
+              --num-trainers 1 \
+              --num-servers 1 \
+              --part-config <PATH_TO_GRAPH_DATA> \
+              --cf <PATH_TO_CONFIG> \
+
+Run Model Inference
+~~~~~~~~~~~~~~~~~~~~
+You can use the same command line `graphstorm.run.gs_multi_task_learning` to run inference as following:
```

Comment:
"command line `graphstorm.run.gs_multi_task_learning` to" -> "command line `graphstorm.run.gs_multi_task_learning` with an additional argument `--inference` to"